lib/config.js

```diff
@@ -86,6 +87,12 @@ function loadConfig(filePath) {
             config = require(filePath);
         }
 
+        if (config.rules) {
```

Comment:
I have to debug through the code to be sure, but it feels like you are going to run validations multiple times on the same rule. Once here, and once in eslint.js, and if eslint called programmatically, an extra time in CLI-engine.